cmake/Modules/FindAWSSDK_EP.cmake

```diff
@@ -105,7 +105,12 @@ if (AWSSDK_FOUND)
   list(APPEND AWS_LINKED_LIBS aws-c-common
                               aws-c-event-stream
                               aws-checksums)
+
   foreach (LIB ${AWS_LINKED_LIBS})
+    if (NOT ${LIB} MATCHES "aws-*")
```

Comment:
I think we still need to execute most of this loop, right? Just not the `install_target_libs` line?

Reply:
For context, `AWS_LINKED_LIBS` is `aws-cpp-sdk-s3;aws-cpp-sdk-core;pthread;crypto;ssl;z;curl;aws-c-common;aws-c-event-stream;aws-checksums`

Is there a case where it is necessary to add the `AWSSDK::<lib>` targets for a non-AWS library? (FWIW, my local static build was successful, and the CI for this PR looks ok)

I just added some debugging statements and it doesn't seem that `find_library` finds anything for those anyway, at least in a static+full-deps build.

```
-- Found AWS lib: crypto (AWS_FOUND_crypto-NOTFOUND)
CMake Warning at cmake/Modules/FindAWSSDK_EP.cmake:128 (message):
  SKIPPING crypto : AWS_FOUND_crypto-NOTFOUND
Call Stack (most recent call first):
  tiledb/CMakeLists.txt:272 (find_package)


-- Found AWS lib: ssl (AWS_FOUND_ssl-NOTFOUND)
CMake Warning at cmake/Modules/FindAWSSDK_EP.cmake:128 (message):
  SKIPPING ssl : AWS_FOUND_ssl-NOTFOUND
Call Stack (most recent call first):
  tiledb/CMakeLists.txt:272 (find_package)


-- Found AWS lib: z (AWS_FOUND_z-NOTFOUND)
CMake Warning at cmake/Modules/FindAWSSDK_EP.cmake:128 (message):
  SKIPPING z : AWS_FOUND_z-NOTFOUND
Call Stack (most recent call first):
  tiledb/CMakeLists.txt:272 (find_package)


-- Found AWS lib: curl (AWS_FOUND_curl-NOTFOUND)
CMake Warning at cmake/Modules/FindAWSSDK_EP.cmake:128 (message):
  SKIPPING curl : AWS_FOUND_curl-NOTFOUND
Call Stack (most recent call first):
  tiledb/CMakeLists.txt:272 (find_package)
```

Reply:
Oh, I'm sorry, I misread the conditional. It's good as it is.